src/nprintml/pipeline.py

```diff
@@ -402,6 +410,47 @@ def merge(self, results):
 
             setattr(self.results, key, value)
 
+    def pre_satisfy(self, *fields):
+        """Remove from the pipeline those steps which provide the given
+        pre-satisfied `fields` of results.
+
+        This is a generator method, iteration of which produces the
+        values of removed steps' `group_parser` attribute, for the
+        purpose of updating the CLI to match (e.g. to ensure that
+        arguments of ejected steps are not required).
+
+        The pipeline is considered in reverse. Steps which provide no
+        more than those fields which have been pre-satisfied are
+        removed. Subsequently, fields which were required by only these
+        ejected steps are considered pre-satisfied as well (by cascade).
+
+        """
+        if not fields:
+            return
+
+        satisfied = set(fields)
+
+        for step in reversed(self):
+            if satisfied.issuperset(step.provides):
+                # we no longer need this step to provide what's already
+                # been satisfied -- remove it
+                self.remove(step)
+
+                # without the step in the pipeline its parser interface
+                # is no longer important/required -- share this with the
+                # caller (rather than attempt to handle here).
+                yield getattr(step, 'group_parser', None)
+
+                # let's also pretend therefore that all of that step's
+                # requirements have been satisfied -- IFF no other steps
+                # have the same requirement(s)
+                step_required = set(step.__requires__)
+                remaining_required = itertools.chain.from_iterable(step.__requires__
+                                                                   for step in self)
+                cascade_satisfied = step_required.difference(remaining_required)
+
+                satisfied.update(cascade_satisfied)
```

Comment:
This stuff could perhaps still be made "smarter" – but, it's more than we need for now.